File: Structural/Adapter/Python/mp3_player.py
# mp3_player.py
from media_player import MediaPlayer
from media_adapter import MediaAdapter

class MP3Player(MediaPlayer):
  def play(self, audio_type: str, file_name: str):
    if audio_type.lower() == "mp3":
      print(f"Playing MP3 file: {file_name}")
    elif audio_type.lower() in ["mp4", "vlc"]:
      adapter = MediaAdapter(audio_type)
      adapter.play(audio_type, file_name)
    else:
      print(f"Invalid media type: {audio_type}")

File: Structural/Adapter/JavaScript/mp3Player.js
// mp3Player.js
const MediaPlayer = require("./mediaPlayer");
const MediaAdapter = require("./mediaAdapter");

class MP3Player extends MediaPlayer {
  play(audioType, fileName) {
    if (audioType.toLowerCase() === "mp3") {
      console.log(`Playing MP3 file: ${fileName}`);
    } else if (audioType.toLowerCase() === "mp4" || audioType.toLowerCase() === "vlc") {
      const adapter = new MediaAdapter(audioType);
      adapter.play(audioType, fileName);
    } else {
      console.log(`Invalid media type: ${audioType}`);
    }
  }
}

module.exports = MP3Player;
